api/java/java_model.go

```diff
@@ -51,6 +51,51 @@ func RequestToArtifact(vars map[string]string) (ja Artifact) {
 	}
 }
 
+// StorageIdentifierToArtifact : Convert a storage identifier into an Artifact.
+func StorageIdentifierToArtifact(id storage.Identifier) (ja Artifact) {
+
+	// Remove the initial slash if there is one (by mistake)
+	sanitized := strings.TrimPrefix(id.Key, id.Separator)
+
+	// Split the path into a set of
```

Comment:
In order for the parsing to work, we can't have a slash (or whatever separator) at the start (like, a root dir) otherwise it will append an extra dot to the group. I get the feeling that it'll happen (by mistake or not) so I thought I'd start by sanitizing the input.